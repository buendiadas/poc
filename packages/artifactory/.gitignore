/src/codegen
/src/abi
/src/bytecode
