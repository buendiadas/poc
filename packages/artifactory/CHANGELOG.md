# @crestproject/artifactory

## 0.0.73

### Patch Changes

- b19f1b2: updating build
- Updated dependencies [b19f1b2]
  - @crestproject/crestproject@0.0.73

## 0.0.72

### Patch Changes

- 505f030: export signer
- Updated dependencies [505f030]
  - @crestproject/crestproject@0.0.72

## 0.0.71

### Patch Changes

- a738ec7: fixing babel setup
- Updated dependencies [a738ec7]
  - @crestproject/crestproject@0.0.71

## 0.0.70

### Patch Changes

- 2aa526a: fixing package configs
- Updated dependencies [2aa526a]
  - @crestproject/crestproject@0.0.70
