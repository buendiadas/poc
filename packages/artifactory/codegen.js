const fs = require('fs-extra');
const path = require('path');
const { utils } = require('ethers');
const { formatOutput, generateContract } = require('@crestproject/codegen');

function generate(file) {
  const name = path.basename(file, '.json');
  const contract = require(file);
  const destination = path.resolve(__dirname, `src/codegen/${name}.ts`);

  const abi = new utils.Interface(contract.abi);
  const output = generateContract(name, contract.bytecode, abi);
  const formatted = formatOutput(output);

  fs.outputFileSync(destination, formatted);
}

generate('./artifacts/contracts/AmIRichAlready.sol/BasicToken.json');
generate('./artifacts/contracts/AmIRichAlready.sol/AmIRichAlready.json');
