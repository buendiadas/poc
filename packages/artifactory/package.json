{
  "name": "@crestproject/artifactory",
  "version": "1.0.1",
  "private": true,
  "main": "dist/artifactory.cjs.js",
  "scripts": {
    "compile": "hardhat compile",
    "postcompile": "yarn codegen",
    "codegen": "crestproject codegen \"artifacts/contracts/**/!(*.dbg).json\" src/codegen"
  },
  "dependencies": {
    "@babel/runtime": "^7.12.1"
  },
  "peerDependencies": {
    "@crestproject/crestproject": "^0.1.1"
  },
  "devDependencies": {
    "@crestproject/codegen": "^0.1.1",
    "@crestproject/crestproject": "^0.1.1",
    "@openzeppelin/contracts": "^3.2.0",
    "hardhat": "^2.0.2"
  }
}
