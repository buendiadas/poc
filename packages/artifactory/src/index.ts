export { BasicToken } from './codegen/BasicToken';
export { AmIRichAlready } from './codegen/AmIRichAlready';
