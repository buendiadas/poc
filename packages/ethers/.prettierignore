# TODO: Remove this again once prettier supports typescript v4
src/tests/contracts
src/doppelganger.ts
