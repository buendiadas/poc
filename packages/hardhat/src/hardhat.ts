import deepmerge from 'deepmerge';
import { HardhatArguments, HardhatConfig, HardhatNetworkConfig } from 'hardhat/types';
import { HardhatContext } from 'hardhat/internal/context';
import { Environment } from 'hardhat/internal/core/runtime-environment';
import { loadConfigAndTasks } from 'hardhat/internal/core/config/config-loading';
import { getEnvHardhatArguments } from 'hardhat/internal/core/params/env-variables';
import { HARDHAT_NETWORK_NAME } from 'hardhat/internal/constants';
import { HARDHAT_PARAM_DEFINITIONS } from 'hardhat/internal/core/params/hardhat-params';

export async function hardhat(network: Partial<HardhatNetworkConfig> = {}) {
  if (!HardhatContext.isCreated()) {
    HardhatContext.createHardhatContext();
  }

  const context = HardhatContext.getHardhatContext();
  const config = deepmerge<HardhatConfig>(loadConfigAndTasks(), {
    networks: {
      [HARDHAT_NETWORK_NAME]: network,
    } as any,
  });

  const extenders = context.extendersManager.getExtenders();
  const args = deepmerge<HardhatArguments>(getEnvHardhatArguments(HARDHAT_PARAM_DEFINITIONS, process.env), {
    network: HARDHAT_NETWORK_NAME,
    emoji: false,
    help: false,
    version: false,
  });

  return new Environment(config, args, {}, extenders);
}
