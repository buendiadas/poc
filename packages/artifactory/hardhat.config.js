require('@crestproject/hardhat/codegen');

module.exports = {
  solidity: {
    version: '0.6.8',
  },
  codeGenerator: {
    enabled: true,
    clear: true,
    bytecode: {
      path: './src/bytecode',
    },
    abi: {
      path: './src/abi',
    },
    typescript: {
      path: './src/codegen',
    },
  },
};
