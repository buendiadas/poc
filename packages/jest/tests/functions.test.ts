import { HardhatProvider } from '@crestproject/hardhat';
import { BasicToken } from '@crestproject/artifactory';
import { utils } from 'ethers';

async function snapshot(provider: HardhatProvider) {
  const deployer = await provider.getSignerWithAddress(0);
  const someone = await provider.getSignerWithAddress(1);
  const signer = await provider.getSignerWithAddress(2);
  const token = await BasicToken.deploy(signer, utils.parseEther('100'));

  return {
    deployer,
    someone,
    token,
  };
}

describe('functions', () => {
  it('toMatchGasSnapshot', async () => {
    const { token, someone } = await provider.snapshot(snapshot);

    const tx = token.transfer(someone, '456');
    await expect(tx).resolves.toMatchGasSnapshot(`51319`);
  });

  it('toCostLessThan', async () => {
    const { token, someone } = await provider.snapshot(snapshot);

    const tx = token.transfer(someone, '456');
    await expect(tx).resolves.toCostLessThan('51319');
  });
});
