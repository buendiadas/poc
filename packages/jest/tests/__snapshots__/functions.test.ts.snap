// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions toMatchGasSnapshot 1`] = `51319`;
